IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework.Tests/TestCaseTests/TestFailureTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using IntelliTect.TestTools.TestFramework.Tests.TestData.Dependencies;
+using IntelliTect.TestTools.TestFramework.Tests.TestData.TestBlocks;
+using System;
+using Xunit;
+
+namespace IntelliTect.TestTools.TestFramework.Tests.TestCaseTests
+{
+    public class TestFailureTests
+    {
+        [Fact]
+        public void TestFailureThrowsImmediatelyWithOriginalException()
+        {
+            TestBuilder builder = new();
+            TestCase tc = builder
+                .AddDependencyInstance(false)
+                .AddDependencyInstance("Testing")
+                .AddTestBlock<ExampleTestBlockWithBoolReturn>()
+                .AddTestBlock<ExampleTestBlockWithMultipleDependencies>(1)
+                .Build();
+
+            var ex = Assert.Throws<TestCaseException>(() => tc.Execute());
+            Assert.False(tc.Passed);
+            Assert.NotNull(ex.InnerException);
+            Assert.Equal(typeof(DivideByZeroException), ex.InnerException!.GetType());
+            Assert.Equal("test failure", ex.InnerException.Message, ignoreCase: true);
+        }
+
+        [Fact]
+        public void DependencyWithMissingDependencyThrowsOriginalError()
+        {
+            TestCase tc = new TestBuilder()
+                .AddDependencyService<AlwaysThrow>(new ExampleFactory().Throws)
+                .AddDependencyService<SomeDependency>()
+                .AddTestBlock<SomeTestBlock>()
+                .Build();
+
+            var ex = Assert.Throws<TestCaseException>(() => tc.Execute());
+            Assert.False(tc.Passed);
+            Assert.NotNull(ex.InnerException);
+            Assert.Equal(typeof(InvalidOperationException), ex.InnerException!.GetType());
```

Comment:
A better type check is:
```suggestion
            Assert.IsType<InvalidOperationException>(ex.InnerException);
```